Use output directory under wx directory in the build tree

vadz opened the pull request:
CMAKE_CURRENT_BINARY_DIR is the same as CMAKE_BINARY_DIR when building wx directly, but they can be different when including this makefile from a super-project using add_subdirectory() and it makes more sense to use the former.

This also fixes wx-config symlink created in such build, as it the link was broken before because it was created under CMAKE_CURRENT_BINARY_DIR but pointed to a file in a different CMAKE_BINARY_DIR.

---

This is another CMake fix that I'm relatively confident in, for the reasons described above, but a confirmation from @TcT2k and/or @MaartenBent would still be welcome. TIA!

vadz committed:
Use output directory under wx directory in the build tree

CMAKE_CURRENT_BINARY_DIR is the same as CMAKE_BINARY_DIR when building
wx directly, but they can be different when including this makefile from
a super-project using add_subdirectory() and it makes more sense to use
the former.

This also fixes wx-config symlink created in such build, as it the link
was broken before because it was created under CMAKE_CURRENT_BINARY_DIR
but pointed to a file in a different CMAKE_BINARY_DIR.

MaartenBent commented:
Maybe we should also replace the other usage of `CMAKE_BINARY_DIR` in `lib/webview/CMakeLists.txt` with `wxBINARY_DIR`.

There is also `CMAKE_CURRENT_SOURCE_DIR` and `CMAKE_CURRENT_BINARY_DIR` in `config.cmake`, for consistency those could also use `wxSOURCE_DIR` and `wxBINARY_DIR`.

vadz committed:
Download WebView2 under current, and not top level, directory

This is done for consistency with the parent commit and to keep things a
bit tidier, even if it doesn't really change anything.

vadz committed:
Use wxSOURCE_DIR and wxBINARY_DIR in config.cmake for consistency

We prefer using these variables rather than CMake ones directly
elsewhere, so do it here too for consistency.

No real changes.

vadz commented:
I'm actually not really sure why do we use our own variables instead of the CMake standard ones, but I've changed these occurrences too for consistency as you suggested now.

The only remaining occurrences of the standard variables are now in Cotire module which I'm not going to touch (BTW, I've tried to use `-DwxBUILD_PRECOMP=COTIRE` just to see what it would do, but it only resulted in errors...).
